### ISSUE TITLE
Question about dataset

### ISSUE THREAD

**Issue by Ca-ressemble-a-du-fake** [DigitalPhonetics/IMS-Toucan#56]
Hi,

In the [paper](https://arxiv.org/pdf/2210.12223.pdf) 5 minutes of speech are used to train on a new language. But for finetuning the Meta model on an already seen language (say French) is it worth it providing hours of single speaker audio ? I mean will the quality of the model improve when providing more than 5 minutes of audio ? I tried to provide a home made 75 minute dataset but I still could not recognize the speaker in the generated audio after finetuning up to 120k steps, although the prosody was awesome!

And regarding speaker transfer (see interactive demo) because you output 44kHz audio, should the input audio (reference audio) also be 44 kHz ? I tried with 16 kHz audios but could not recognize the reference speaker from the generated output, although as mentioned earlier prosody was pretty good.

So to summarize : 

1. how long the dataset should be ?
2. which framerate should it have ?
3. how long should each sample be ? I understand from this [chat](https://github.com/DigitalPhonetics/IMS-Toucan/issues/25#issuecomment-1124410552) that adding longer samples enhanced the quality of the model. But to what extent, should I add 20 s, 30 s, 60 s audios in my dataset ?

Thank you very much for your help :smile:

**Comment by Flux9665** [comment]
1. For datasets, it's best to use cleaner data than more data, but 75 minutes should be more than enough to learn the new voice, so it's strange that it doesn't work.
2. The sampling rate of the reference does not matter, it is detected automatically and resampled to 16kHz if you change the speaker embedding using the filepath method.
3. The length of the samples is not very important. Longer samples might cause your model to run out of memory, so I usually remove anything over 20 seconds. If your data consists of only very short samples (e.g. one word per sample), then the prosody cannot be learned from that data, but as long as you have a couple of full sentences or sentence fragments in your data, it should be fine. TTS datasets usually have an average sample duration of 6s.

**Comment by Ca-ressemble-a-du-fake** [comment]
That's very clear instructions thank you! I will rework my dataset to make it cleaner but I feel relieved as it follows your third point and only contains sentences.

Can I ask you what you think of that spectrogram ? I see dark stripes but some of them are nearly horizontal and it is all a bit blurry (I followed your [post](https://github.com/DigitalPhonetics/IMS-Toucan/issues/19#issuecomment-1101617332) to know what to look at). So would more training steps help (the picture was taken at step 180k with bs 32 if that matters)?

![180104](https://user-images.githubusercontent.com/91517923/202358243-49c898a6-8a6a-4af4-a484-96dec6afd94a.png)

**Comment by Flux9665** [comment]
I think this spectrogram looks very good! It looks like it is probably a male speaker. Because male speakers have generally a lower fundamental frequency, the lines are pretty close together in the lower part. But it still looks pretty clean. If you were generating a spectrogram with the embedding of a female speaker (if there's female speaker(s) in your data), then it would probably look even nicer. But yeah, overall I think the model is well trained.

**Comment by Ca-ressemble-a-du-fake** [comment]
You rock Florian! Your explanations are very clear, thank you! Indeed this is a male voice. I find the model much deeper than the original voice. So I will try another cleaner dataset. Actually the model trained on a female dataset looks like this (I show it to illustrate the second part of your answer as it could help other readers) :
![200430](https://user-images.githubusercontent.com/91517923/202784305-6f566c55-e9be-4d1c-b97c-445b526fa0c9.png)

I have another question still pertaining to dataset creation that I encountered while creating a new one. Should I add starting and trailing silences of 50 ms to each sample in the dataset ? 

I ask this question because when I started using Toucan and ran the training script, some warnings appeared during the preprocessing steps. Unfortunately I could not see those warnings again. These warnings were about missing silences or something around that topic but I cannot recollect exactly.

**Comment by Flux9665** [comment]
We will soon release updated models with a much better trained embedding function, so I hope the speaker adaptation will be better then. There are still some things we are trying out, but it's already a lot better than it was.

About the silences: The preprocessing automatically removes silences in the front and in the end. It actually does not matter too much if they are there or not, because the aligner figures out their duration anyway. Silences in front and end were always a big problem if the duration is learned implicitly, in models like e.g. Tacotron. Now that we use an aligner, this problem is pretty much solved. 

This part of the preprocessing that automatically detects silences requires a minimum duration for an audio. So if you have super short audios in your data, the preprocessor prints a warning, but it's safe to ignore.

**Comment by Ca-ressemble-a-du-fake** [comment]
Ok pretty clear as usual, thank you very much!

**Comment by kin0303** [comment]
> 2\. The sampling rate of the reference does not matter, it is detected automatically and resampled to 16kHz if you change the speaker embedding using the filepath method.

How to use the speaker embedding using the filepath method?
